src/rate_limiter/src/lib.rs

```diff
@@ -101,6 +87,7 @@ pub struct TokenBucket {
     initial_one_time_burst: u64,
     // Complete refill time in milliseconds.
     refill_time: u64,
+    refill_one_ns: u64,
```

Comment:
LGTM. Can you move the PR explanation into the commit body?

Reply:
I've moved the explanation to the commit body.

Reply:
Could you suggest an what you are looking for here?

Reply:
Something along the lines:

> Our implementation of `TokenBucket` can dead-lock because of...
>
> Change the implementation in such and such way to solve this.

Reply:
I've update the commit message to match the new implementation.